Source/WebCore/Modules/webcodecs/WebCodecsAudioDecoder.cpp

```diff
@@ -177,17 +178,15 @@ ExceptionOr<void> WebCodecsAudioDecoder::flush(Ref<DeferredPromise>&& promise)
         return Exception { ExceptionCode::InvalidStateError, "AudioDecoder is not configured"_s };
 
     m_isKeyChunkRequired = true;
-    m_pendingFlushPromises.append(promise.copyRef());
-    m_isFlushing = true;
-    queueControlMessageAndProcess([this, clearFlushPromiseCount = m_clearFlushPromiseCount] {
+    m_pendingFlushPromises.append({ *this, WTFMove(promise) });
+    queueControlMessageAndProcess({ *this, [this, clearFlushPromiseCount = m_clearFlushPromiseCount] {
         m_internalDecoder->flush([this, clearFlushPromiseCount] {
             if (clearFlushPromiseCount != m_clearFlushPromiseCount)
                 return;
 
-            m_pendingFlushPromises.takeFirst()->resolve();
-            m_isFlushing = !m_pendingFlushPromises.isEmpty();
+            m_pendingFlushPromises.takeFirst().resolve();
```

Comment:
It is ok to use a pending activity instead of m_isFlushing.
But I am not sure why we need WebCodecsFlushPromise, instead of taking a pending activity in `m_internalDecoder->flush` callback.

Reply:
That also works and would mirror what we do for `m_internalDecoder->decode`.